### ISSUE TITLE
Add Workflow Editor

### ISSUE THREAD

**Issue by douginator2000** [qiime/qiime_web_app#133]
This is just a brainstorm for now, but could have awesome applications. 

A workflow editor for data processing and analysis could be very awesome. This would be a drag-drop interface. Objects would represent items (e.g. metadata, sequence files, or processing objects), and can be connected in a pipeline. Objects would be individually configurable (e.g. sequence file - set split_library options, etc...)... 

Use-case for data upload:

User drags sequence file onto workflow - upload happens, options can be set per-sequence file if necessary (split_library configuration). Would be largely locked down probably, at least enough to keep database data consistent. Metadata files dragged onto interface - would perform validation and allow you to fix errors. Once multiple objects exist, they would be intelligently linked (i.e. sequence files and metadata would be linked to an end-note that represents the process job).

Think through analysis interface.

**Comment by rob-knight** [comment]
Yeah this could be spectacular. How much of this if any does galaxy and/or Henry's workflow tools provide already and can we leverage?

Rob

On Feb 19, 2013, at 1:32 PM, "Doug Wendel" <notifications@github.com<mailto:notifications@github.com>> wrote:

This is just a brainstorm for now, but could have awesome applications.

A workflow editor for data processing and analysis could be very awesome. This would be a drag-drop interface. Objects would represent items (e.g. metadata, sequence files, or processing objects), and can be connected in a pipeline. Objects would be individually configurable (e.g. sequence file - set split_library options, etc...)...

Use-case for data upload:

User drags sequence file onto workflow - upload happens, options can be set per-sequence file if necessary (split_library configuration). Would be largely locked down probably, at least enough to keep database data consistent. Metadata files dragged onto interface - would perform validation and allow you to fix errors. Once multiple objects exist, they would be intelligently linked (i.e. sequence files and metadata would be linked to an end-note that represents the process job).

Think through analysis interface.

**Comment by douginator2000** [comment]
I don't know Galaxy very well - I'll have to look into it. Not sure what Henry's workflow tools are - should check into that as well. I've used a number of 3rd party tools in the past, but those were all for the .NET environment.
